commit 763749702064140f12ca2ba98f102622bab3031e
Author: Radek Novacek <rnovacek@redhat.com>
Date:   Thu Mar 10 16:45:18 2016 +0100

    esx: handle the case when no hostname is returned from esx

diff --git a/virt/esx/esx.py b/virt/esx/esx.py
index f392a33..7dee9ea 100644
--- a/virt/esx/esx.py
+++ b/virt/esx/esx.py
@@ -199,7 +199,11 @@ class Esx(virt.Virt):
                                              hypervisorType=host.get('config.product.name', 'vmware'),
                                              hypervisorVersion=host.get('config.product.version', None)
                                              ))
-            name = '%(config.network.dnsConfig.hostName)s.%(config.network.dnsConfig.domainName)s' % host
+            try:
+                name = '%(config.network.dnsConfig.hostName)s.%(config.network.dnsConfig.domainName)s' % host
+            except KeyError:
+                self.logger.debug("Unable to determine hostname for host '%s'", uuid)
+                name = ''
             facts = {
                 'cpu.cpu_socket(s)': str(host['hardware.cpuInfo.numCpuPackages']),
             }
